scipy/_lib/_testutils.py

```diff
@@ -335,3 +336,38 @@ def load(modname):
 
     # test that the module can be imported
     return load("extending"), load("extending_cpp")
+
+
+def _run_concurrent_barrier(n_workers, fn, *args, **kwargs):
```

Comment:
This function looks nice and concise! Would be nice to mention it in the devguide + briefly explain the relation / contrast to the use of other stdlib tools --- ThreadPoolExecutor, concurrent.futures etc. Am not asking for a grounds up threading tutorial. Am asking for something to follow reading up one. Can this be replaced by, say, ThreadPoolExecutor, is it equivalent, is it better (how?) etc. If it's just one of several ways to skin the cat, great, just state it.

Reply:
`ThreadPoolExecutor` acts as syntactic sugar over the same threading primitives that are being used here. What's special about this function is that the closure to be run by the thread group gets signaled with a barrier, thus ensuring that the threads will call the closure concurrently.

Reply:
This would be great to add to the docs